report_generator_3.py

```diff
@@ -0,0 +1,18 @@
+import subprocess
+
+def generate_report(report_title, content_file):
+    """Generates a PDF report with a user-supplied title."""
+    
+
+    command = f"pandoc {content_file} -o report.pdf --metadata title='{report_title}'"
```

Comment:
Critical Command Injection

The `report_title` variable, which directly takes user input, is used to construct a shell command without proper sanitization. This allows an attacker to inject arbitrary shell commands by manipulating the report title. The use of `shell=True` in `subprocess.run` exacerbates this vulnerability.

```suggestion
    # Construct the command as a list of arguments
    command_args = ["pandoc", content_file, "-o", "report.pdf", "--metadata", f"title={report_title}"]
    # Pass the command as a list and avoid shell=True
    result = subprocess.run(command_args, capture_output=True, text=True)
```